src/gbm/gblinear_model.cc

```diff
@@ -0,0 +1,114 @@
+/*!
+ * Copyright 2019 by Contributors
+ */
+#include <utility>
+#include <limits>
+
+#include "gblinear_model.h"
+#include "xgboost/json_io.h"
+
+namespace xgboost {
+namespace gbm {
+
+class LinearReader : public JsonReader {
+ public:
+  explicit LinearReader(StringView str) : JsonReader(str) {}
+
+  void ParseWeight(GBLinearModelParam const& param, std::vector<float>* p_weights) {
+    size_t n_weights = (param.num_feature + 1) * param.num_output_group;
+    p_weights->resize(n_weights);
+    auto& weights = *p_weights;
+    GetChar('[');
+    for (size_t i = 0; i < n_weights; ++i) {
+      char* end {nullptr};
+      auto beg = raw_str_.c_str() + cursor_.Pos();
+
+      weights[i] = std::strtof(beg, &end);
+
+      auto len = std::distance(beg, static_cast<char const*>(end));
+      for  (decltype(len) j = 0 ; j < len; ++j) { cursor_.Forward(); }
+
+      if (i != n_weights - 1) {
+        GetChar(',');
+      }
+    }
+    GetChar(']');
+  }
+};
+
+void GBLinearModel::Save(Json* p_out) const {
+  using WT = std::remove_reference<decltype(std::declval<decltype(weight)>().back())>::type;
+  using JT = Number::Float;
+  static_assert(std::is_same<WT, JT>::value, "");
+  auto& out = *p_out;
+  out["model_param"] = toJson(param);
+  std::string raw;
+  raw.reserve(
+      weight.size() * std::numeric_limits<Number::Float>::max_digits10 /*floats*/ +
```

Comment:
Note that std::numeric_limits<Number::Float>::max_digits10 is just amount of digits one need to perfect serialization-deserialization without lost of data. The resulting string would be longer because we also need to output dot ('.') and possible minus ('-') for negative numbers if any

So I suggest to add +1 here and even +2 if negative numbers are possible